File: CS/10_StampsAndWatermarks/TextWaterMark.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;
using Spire.Pdf.Tables;

namespace TextWaterMark
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a PDF document and load a file from the disk.
            PdfDocument doc = new PdfDocument();
            doc.LoadFromFile(@"..\..\..\..\..\..\Data\TextWaterMark.pdf");

            // Get the first page from the document.
            PdfPageBase page = doc.Pages[0];

            // Create a tiling brush for drawing a text watermark.
            PdfTilingBrush brush = new PdfTilingBrush(new SizeF(page.Canvas.ClientSize.Width / 2, page.Canvas.ClientSize.Height / 3));

            // Set transparency for the brush graphics.
            brush.Graphics.SetTransparency(0.3f);

            // Save the current state of the brush graphics.
            brush.Graphics.Save();

            // Translate and rotate the brush graphics to position the watermark.
            brush.Graphics.TranslateTransform(brush.Size.Width / 2, brush.Size.Height / 2);
            brush.Graphics.RotateTransform(-45);

            // Draw the text watermark using the brush graphics.
            brush.Graphics.DrawString("Spire.Pdf Demo",
                new PdfFont(PdfFontFamily.Helvetica, 24), PdfBrushes.Violet, 0, 0,
                new PdfStringFormat(PdfTextAlignment.Center));

            // Restore the previous state of the brush graphics.
            brush.Graphics.Restore();

            // Reset the transparency to fully opaque.
            brush.Graphics.SetTransparency(1);

            // Draw a rectangle filled with the tiling brush as the watermark on the page.
            page.Canvas.DrawRectangle(brush, new RectangleF(new PointF(0, 0), page.Canvas.ClientSize));

            // Save the modified PDF document to a file.
            doc.SaveToFile("TextWaterMark.pdf");

            // Close the PDF document.
            doc.Close();

            //Launch the Pdf file
            PDFDocumentViewer("TextWaterMark.pdf");
        }

        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB/10_StampsAndWatermarks/TextWaterMark.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics

Namespace TextWaterMark
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new instance of PdfDocument
            Dim doc As New PdfDocument()

            ' Load the PDF file from a specified path
            doc.LoadFromFile("..\..\..\..\..\..\Data\TextWaterMark.pdf")

            ' Get the first page of the document
            Dim page As PdfPageBase = doc.Pages(0)

            ' Create a PdfTilingBrush with a size of half the width and one-third the height of the page canvas
            Dim brush As New PdfTilingBrush(New SizeF(page.Canvas.ClientSize.Width / 2, page.Canvas.ClientSize.Height / 3))

            ' Set the transparency of the brush's graphics context
            brush.Graphics.SetTransparency(0.3F)

            ' Save the current state of the brush's graphics context
            brush.Graphics.Save()

            ' Translate the origin of the brush's graphics context to the center of the brush's size
            brush.Graphics.TranslateTransform(brush.Size.Width / 2, brush.Size.Height / 2)

            ' Rotate the brush's graphics context by -45 degrees
            brush.Graphics.RotateTransform(-45)

            ' Draw the text "Spire.Pdf Demo" using a specified font, color, and alignment within the brush's graphics context
            brush.Graphics.DrawString("Spire.Pdf Demo", New PdfFont(PdfFontFamily.Helvetica, 24), PdfBrushes.Violet, 0, 0, New PdfStringFormat(PdfTextAlignment.Center))

            ' Restore the previous state of the brush's graphics context
            brush.Graphics.Restore()

            ' Reset the transparency of the brush's graphics context
            brush.Graphics.SetTransparency(1)

            ' Draw a rectangle on the page canvas using the brush as a background
            page.Canvas.DrawRectangle(brush, New RectangleF(New PointF(0, 0), page.Canvas.ClientSize))

            ' Specify the output file name for the modified PDF document
            doc.SaveToFile("TextWaterMark.pdf")

            ' Close the PDF document
            doc.Close()

            ' Launch the Pdf file
            PDFDocumentViewer("TextWaterMark.pdf")
        End Sub
        Private Sub PDFDocumentViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try
        End Sub

    End Class
End Namespace
